tests/test_runlib.py

```diff
@@ -726,6 +819,43 @@ def test_normalize_line_endings(self):
       for path in paths:
         os.remove(path)
 
+  def test_nonexistent_directory(self):
+    """Test record stop with nonexistent metadata directory"""
+    expected_error = IOError if sys.version_info < (3, 0) \
+        else FileNotFoundError
+    with self.assertRaises(expected_error):
+      in_toto_record_start(self.step_name, [], self.key)
+      in_toto_record_stop(self.step_name, [], self.key,
+          metadata_directory='nonexistentDir')
+
+  @unittest.skipIf(sys.version_info < (3, 0),
+      "NotADirectoryError is python3 only")
+  def test_not_a_directory(self):
+    """Test record stop, passed metadata directory is not a dir"""
+    fd, path = tempfile.mkstemp()
+    os.write(fd, b"hello in-toto")
+    os.close(fd)
+    with self.assertRaises((NotADirectoryError, FileNotFoundError)):
```

Comment:
Just to confirm and keep things straight in my head; this is because on Windows we're seeing `FileNotFoundError` instead of `NotADirectoryError` correct? Can we follow that down and see where the difference is? Is it `os.path.join`?

Reply:
Yes, this is because on Windows's we're seeing `FileNotFoundError` instead of `NotADirectoryError`. But I don't understand what do you mean by `os.path.join`. Could you explain a bit more? @adityasaky

Reply:
I'm curious exactly which operation is raising this error (either way). As I recall, the value of `metadata_directory` is used in `os.path.join` so is that a difference in implementation there, between Windows and the rest :tm:?

Reply:
Yes, you are right. @adityasaky  
I change ` filename = os.path.join(metadata_directory , filename)` to `filename = metadata_directory + filename`, and this will raise `NotADirectoryError`.

Reply:
This is very interesting. I'd have more expected it to be an issue with `open()` rather than `join`, because I didn't think `join` cared about what contents it was stringing together. Can you dump the difference in the `filename` strings when you obtain it through `os.path.join` and `metadata_directory + filename`?

Reply:
Do you mean dump the `filename` strings to file and compare them? If so, the result strings look the same to me.

Reply:
I see. For now, let's do a `skipIf` and move on. Perhaps you can open an issue for this particular case? :)

Reply:
Correction: I had a brianfart, let's `skipIf` the `PermissionError` test, and retain both exceptions here (with an explanation).